File: sophomore/CE3005/alg/classic/02_Prime/Prime.java
package Prime;

public class Prime {
    public boolean primeCheck1(int n) {
        for (int i = 2; i < n; i++) {
            if (n % i == 0) {
                return false;
            } 
        }
        return true;
    }

    public boolean primeCheck2(int n) {
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (n % i == 0) {
                return false;
            } 
        }
        return true;
    }

    public static void main(String[] Args) {
    
    }
}

File: sophomore/CE3005/alg/classic/02_Prime/Prime.c
#include "stdio.h"
#include "math.h"

int primeCheck1(int);
int primeCheck2(int);

int primeCheck1(int n) {
    for (int i = 2; i < n; i++) {
        if (n % i == 0) {
            return 0;
        } 
    }
    return 1;
}

int primeCheck2(int n) {
    for (int i = 2; i <= sqrt(n); i++) {
        if (n % i == 0) {
            return 0;
        } 
    }
    return 1; 
}